### PR TITLE
Enable SparseNerf depth ranking loss to be used with GT/Sensor depth

### PR DESCRIPTION
**Pull request by alexsax**
PR #2066 adds a very nice ranking-based loss that can be used with non-metric depth. The ranking loss can also be used with metric depth, and seems to provide smoother results on the replica dataset (image below). 

This PR is a minimal set of changes that allows the ranking loss to be used with depth read from disk (usually GT/sensor depth).

image

### CODE REVIEW

**Review by ethanweber — COMMENTED** [review]
These changes look great! Can approve once it's merged again with main and passing the checks.

**Review by AdamRashid96 — APPROVED** [review]
LGTM!

**Pull request merged**
